swift/swift.py

```diff
@@ -221,30 +287,50 @@ def _get_argparser():
     return parser
 
 
-def _read_setup_file(setup_path: str):
-    """Reads set-up information about app and bus from set-up file.
+def _read_setup_file(setup_path: str) -> tuple[Mapping[str, AppInfo], Mapping[str, BusInfo]]:
+    """Reads set-up information about app and bus from a JSON file.
+
+    The JSON file content should have the following structure:
+
+      {
+        "app": {
+          "app_name_0": {app_info_0},
+          ...
+        },
+        "bus": {
+          "bus_name_0": {bus_info_0},
+          ...
+        }
+      }
 
+    See AppInfo and its parse() for app_info_* structure.
+    See BusInfo and its parse() for bus_info_* structure.
+      
     Args:
         setup_path: A path of set-up file.
 
     Returns:
-        dict: A dictionary containing set-up environment about app and bus.
-          For details, see Swift.__init__().
+        (app_infos, bus_infos): Dictionaries of set-up information about app and bus.

```

Comment:
Umm, isn't it a tuple?

Reply:
Oh, I meant (a tuple of) dictionaries. Does it look ambiguous or misleading? If so, I will change the expression.

Reply:
I can understand what you intend, but for newcomers, changing the expression looks better.

Reply:
Alright, I changed the expression!